### ISSUE TITLE
art_urls.json is filled with null entries

### ISSUE THREAD

**Issue by bramtechs** [realoksi/foobar2000-catbox#15]
Sometimes when listening to an album with a cover. The upload supposedly fails and null is placed into ```art_urls.json```.
These entries need to be removed in order to for the upload to be tried again.

snippet of art_urls.json with affected entries
```json
{
  "Matthew S Burns|Opus Magnum Original Soundtrack": "http://ia800403.us.archive.org/8/items/mbid-35cf6285-c58f-430e-9f06-53db35724252/mbid-35cf6285-c58f-430e-9f06-53db35724252-18555739090_thumb1200.jpg",
  "Matthew S. Burns|Opus Magnum Original Soundtrack": "http://ia600403.us.archive.org/8/items/mbid-35cf6285-c58f-430e-9f06-53db35724252/mbid-35cf6285-c58f-430e-9f06-53db35724252-18555739090_thumb1200.jpg",
  "POPHOLIC|龍が如く0 誓いの場所 オリジナルサウンドトラック(Side B)": "http://ia803105.us.archive.org/13/items/mbid-a4d524fe-ada0-46d1-a3b6-9f7dfcc55b2f/mbid-a4d524fe-ada0-46d1-a3b6-9f7dfcc55b2f-36042971363_thumb1200.jpg",
  "Rafał Samborski|Golf Peaks OST": null,
  "The Living Tombstone|Five Nights at Freddy's": "https://files.catbox.moe/jxcupm.jpeg",
  "Victor Machado|Kerbal Space Program Full Soundtrack": null,
  "Victor Machado|Kerbal Space Program Original Soundtrack": null,
  "ZENTA|龍が如く0 誓いの場所 オリジナルサウンドトラック Side A": "http://ia600900.us.archive.org/20/items/mbid-032dea35-ec29-4cca-9d97-be4821ec6905/mbid-032dea35-ec29-4cca-9d97-be4821ec6905-22522786521_thumb1200.jpg",
  "brain.fm|?": "https://files.catbox.moe/4dnkaf.png",
  "ぬいぐるみクレヨン Lush Crayon|?": null,
}
```

**Comment by bramtechs** [comment]
I don't know where this program saves log files when I set it like this foobars settings. Where to find these could also be mentioned in the README file. :)
![image](https://github.com/user-attachments/assets/977338e1-59d7-4065-be3f-de114c108c1c)

**Comment by realoksi** [comment]
> Sometimes when listening to an album with a cover. The upload supposedly fails and null is placed into `art_urls.json`. These entries need to be removed in order to for the upload to be tried again.
> ...

Yeah, this is a problem with the foo_discord_rich plugin itself. I've been able to reproduce `null` entries when using other uploaders and causing them to fail by blocking internet access. User [kam106](https://github.com/kam106) reported it in this closed issue #11, and on the main plugin repository under stale issue [#69](https://github.com/TheQwertiest/foo_discord_rich/issues/69). There's not really anything I can do about this, hopefully it'll get fixed.

> I don't know where this program saves log files when I set it like this foobars settings. Where to find these could also be mentioned in the README file. :) ...

The uploader doesn’t write to any log files; it only sends the link of the uploaded image or error messages directly to the plugin through `stdout`. As far as I know, the plugin also doesn't write to any log files but I could be wrong about that. I do know it can display messages/errors in the console. You can open the console by clicking "View" in the menu bar and selecting "Console" from the dropdown menu in fb2k. Any failure **should** pop up here with a reason why.

#

As for the issue itself causing the upload to fail, can you check the console output to see if there's an error being reported when trying to upload these images? Another issue here #13 mentions that not only are their artworks failing to upload, but there are also no errors being outputted. I was unable to reproduce this.
